src/test/scala/org/broadinstitute/dsde/firecloud/dataaccess/MockSamDAO.scala

```diff
@@ -21,9 +22,14 @@ class MockSamDAO extends SamDAO {
     WorkbenchGroupName("TARGET-dbGaP-Authorized") -> Set(WorkbenchEmail("target-linked"), WorkbenchEmail("target-linked-expired"), WorkbenchEmail("tcga-and-target-linked"), WorkbenchEmail("tcga-and-target-linked-expired"))
   )
 
-  override def registerUser(implicit userInfo: WithAccessToken): Future[RegistrationInfo] = enabledUserInfo
+  override def registerUser(termsOfService: Option[String])(implicit userInfo: WithAccessToken): Future[RegistrationInfo] =
+    if (termsOfService.contains(termsOfServiceUrl)) enabledUserInfo
+    else Future.failed(new FireCloudExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.Forbidden,
+      s"You must accept the Terms of Service in order to register.")))
 
-  override def getRegistrationStatus(implicit userInfo: WithAccessToken): Future[RegistrationInfo] = enabledUserInfo
+  override def getRegistrationStatus(implicit userInfo: WithAccessToken): Future[RegistrationInfo] =
+    if (userInfo.accessToken.token.equals("new")) unknownUserInfo
+    else enabledUserInfo
```

Comment:
is this `if` condition exercised? I see that `RegisterApiServiceSpec` has two tests that send a token containing "new" - but those tests only assert on the http response status, they don't look at the RegistrationInfo details.

Reply:
Yes, it's exercised- when the "unknownUserInfo" is returned, this drives Orch to call Sam (in this case MockSam) to create a new user. If the ToS URL isn't included in the request, than mockSam returns an exception. However, if the token isn't "new", MockSam will return that the user is already known, so Orch won't try to create the user, and no exception will be thrown if there's no ToS parameter.